packages/core/src/modules/presentation-exchange/PresentationExchangeService.ts

```diff
@@ -57,6 +60,50 @@ export class PresentationExchangeService {
     return selectCredentialsForRequest(presentationDefinition, credentialRecords, holderDids)
   }
 
+  public validatePresentationDefinition(presentationDefinition: PresentationDefinition) {
+    const validation = PEX.validateDefinition(presentationDefinition)
+    const errorMessages = this.formatValidated(validation)
+    if (errorMessages.length > 0) {
+      throw new PresentationExchangeError(
```

Comment:
Would it be possible to add these error messages to the error class as a custom property? Then they could be extracted on a higher level for more control

Reply:
Yes, good idea